### ISSUE TITLE
[FEATURE]: Transition blue_localization to marine_localization and independent drivers

### ISSUE THREAD

**Issue by evan-palmer** [Robotic-Decision-Making-Lab/blue#165]
### Feature Type

Adding new functionality to the BlueROV2 driver

### Problem Description

This is a book-keeping issue to track the transition of the blue_localization package to [marine_localization](https://github.com/Robotic-Decision-Making-Lab/marine_localization) and to independent drivers for the respective localization sources.

### Feature Description

The following changes will take place in the scope of this issue:

- [ ] Convert dependency on the ArduSub EKF to either [robot_localization](https://github.com/cra-ros-pkg/robot_localization) (which is unfortunately no longer supported by ROS despite it being a pretty nice package or to [Fuse](https://github.com/locusrobotics/fuse) which is now the official localization framework for ROS but has less support for features that underwater vehicles might use
- [ ] Export Qualisys source to its own driver (or find an alternative driver)
- [ ] Export BlueROV2 camera interface to its own driver
- [ ] Export the A50 DVL implementation to its own driver
- [ ] Document the supported drivers
- [ ] Add a demo showing how to use perform localization with the updated system
- [ ] (optional) Implement publishers for some common sensors as ros2_control sensors. This would be similar to the [imu_sensor_broadcaster](https://github.com/ros-controls/ros2_controllers/tree/master/imu_sensor_broadcaster) and [force_torque_sensor_broadcaster](https://github.com/ros-controls/ros2_controllers/tree/master/force_torque_sensor_broadcaster) and should use APL's [marine_msgs](https://github.com/apl-ocean-engineering/marine_msgs) package.

### Alternative Solutions

n/a

**Comment by evan-palmer** [comment]
@amarburg Here is the issue that I have been tracking the localization things that we discussed offline in

**Comment by evan-palmer** [comment]
Adding this here so that I don't forget in the future:

APL has a nice driver for the Oculus sonar [here](https://github.com/apl-ocean-engineering/liboculus). It would be good to integrate support for that and document it.

**Comment by evan-palmer** [comment]
Regarding the motion capture systems, there has been quite a bit of work in the [MOCAP4ROS2](https://github.com/MOCAP4ROS2-Project) organization that can be used and documented

**Comment by evan-palmer** [comment]
Here is a [ROS 2 driver](https://github.com/CentraleNantesRobotics/ping360_sonar) (supports Foxy) for the Ping 360 sonar

**Comment by amarburg** [comment]
> APL has a nice driver for the Oculus sonar [here](https://github.com/apl-ocean-engineering/liboculus). It would be good to integrate support for that and document it.

I _think_ Rakesh was working on a ROS2 version of the Oculus driver this summer (?)

**Comment by evan-palmer** [comment]
I'm currently testing our new Water Linked DVL driver [here](https://github.com/Robotic-Decision-Making-Lab/waterlinked_dvl/pull/1). Once that is finished, I'll add a section to the docs to list supported drivers.

**Comment by evan-palmer** [comment]
Tracking Fuse support for 3D motion models [here](https://github.com/locusrobotics/fuse/pull/354)

**Comment by evan-palmer** [comment]
Once #292 is merged, I'll open a new PR that deprecates the Qualisys driver.

**Comment by evan-palmer** [comment]
We can document [blue_rov_camera](https://gitlab.com/apl-ocean-engineering/bluerov/blue_rov_camera) as an alternative to the BlueROV camera source used in the project.

**Comment by evan-palmer** [comment]
3D motion models are now supported in Fuse, so it should be possible to use this rather than robot_localization

**Comment by evan-palmer** [comment]
I've been doing some separate work with the invariant EKF, which is quite nice and has a straightforward implementation. It might be worthwhile to try implementing that as opposed to integrating Fuse. I know that Josh Mengelson's group uses the InEKF for their underwater navigation work. I think that integrating `robot_localization` would be a good intermediate step towards this.

**Comment by Manouselis** [comment]
Hey @evan-palmer, the DVL driver builds fine and works like a charm (at least for the `humble` branch, which I tested). Just wanted to mention for future debugging that I am getting a deprecation warning when building:

```
--- stderr: waterlinked_dvl_driver                               
In file included from /home/ros2_ws/src/waterlinked_dvl/waterlinked_dvl_driver/include/waterlinked_dvl_driver/waterlinked_dvl_driver.hpp:35,
                 from /home/ros2_ws/src/waterlinked_dvl/waterlinked_dvl_driver/src/waterlinked_dvl_driver.cpp:21:
/home/ros2_ws/build/waterlinked_dvl_driver/include/waterlinked_dvl_driver_parameters.hpp:1:169: note: ‘#pragma message: #include "waterlinked_dvl_driver_parameters.hpp" is deprecated. Use #include <waterlinked_dvl_driver/waterlinked_dvl_driver_parameters.hpp> instead.’
    1 | d_dvl_driver_parameters.hpp\" is deprecated. Use #include <waterlinked_dvl_driver/waterlinked_dvl_driver_parameters.hpp> instead.")
      |                                                                                                                                   ^

```

**Comment by evan-palmer** [comment]
Thanks for mentioning this @Manouselis! I just submitted a PR that addresses those issues. Should be merged by EOD
